### ISSUE TITLE
Program crash immediately after starting campaign

### ISSUE THREAD

**Issue by Oweoqi** [gophish/gophish#122]
Hey, tried to start campaign after setup and it crashed:

```
127.0.0.1 - - [03/Feb/2016:10:34:22 +0800] "POST /api/campaigns/?api_key=api_key_waz_here HTTP/1.1" 400 74 "http://127.0.0.1:3333/campaigns" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0"
 2016/02/03 10:34:32 main_private.go:23: sql/root/projects/src/github.com/gophish/gophish/models/campaign.go:100477.698µsINSERT INTO "events" ("time","message","campaign_id","email") VALUES (?,?,?,?) [2016-02-03 10:34:32.930666742 +0800 AWST Campaign Created 1 ]
 2016/02/03 10:34:32 worker.go:42: Worker received: test
127.0.0.1 - - [03/Feb/2016:10:34:32 +0800] "POST /api/campaigns/?api_key=api_key_waz_here HTTP/1.1" 201 86506 "http://127.0.0.1:3333/campaigns" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0"
 2016/02/03 10:34:32 worker.go:57: mail: missing phrase
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x647c32]

goroutine 20 [running]:
github.com/gophish/gophish/worker.processCampaign(0xc8207261e0)
	/root/projects/src/github.com/gophish/gophish/worker/worker.go:59 +0x26a2
github.com/gophish/gophish/worker.(*Worker).Start(0xc8201ec030)
	/root/projects/src/github.com/gophish/gophish/worker/worker.go:37 +0x119
created by github.com/gophish/gophish/controllers.init.1
	/root/projects/src/github.com/gophish/gophish/controllers/api.go:28 +0x9f

goroutine 1 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc8201e25ec)
	/root/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc8201e25e0)
	/root/go/src/sync/waitgroup.go:126 +0xb4
main.main()
	/root/projects/src/github.com/gophish/gophish/gophish.go:75 +0x17b

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/root/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 21 [chan receive, 9 minutes]:
database/sql.(*DB).connectionOpener(0xc82024ec80)
	/root/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/root/go/src/database/sql/sql.go:481 +0x336

goroutine 22 [IO wait, 2 minutes]:
net.runtime_pollWait(0x7f3ca46bc378, 0x72, 0xc82000a1c0)
	/root/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820293480, 0x72, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820293480, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820293420, 0x0, 0x7f3ca46bc530, 0xc82032e000)
	/root/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820028ca0, 0x458ee0, 0x0, 0x0)
	/root/go/src/net/tcpsock_posix.go:254 +0x4d
net/http.tcpKeepAliveListener.Accept(0xc820028ca0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/net/http/server.go:2135 +0x41
net/http.(*Server).Serve(0xc82027b260, 0x7f3ca46bc4f8, 0xc820028ca0, 0x0, 0x0)
	/root/go/src/net/http/server.go:1887 +0xb3
net/http.(*Server).ListenAndServe(0xc82027b260, 0x0, 0x0)
	/root/go/src/net/http/server.go:1877 +0x136
net/http.ListenAndServe(0xc82000af00, 0xe, 0x7f3ca46bb460, 0xc82028e5e0, 0x0, 0x0)
	/root/go/src/net/http/server.go:1967 +0x8f
main.main.func1(0xc8201e25e0)
	/root/projects/src/github.com/gophish/gophish/gophish.go:60 +0x671
created by main.main
	/root/projects/src/github.com/gophish/gophish/gophish.go:62 +0x134

goroutine 23 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7f3ca46bc438, 0x72, 0xc82000a1c0)
	/root/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201ef790, 0x72, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201ef790, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8201ef730, 0x0, 0x7f3ca46bc530, 0xc8201d8560)
	/root/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc8201ec0a8, 0xc820055c88, 0x0, 0x0)
	/root/go/src/net/tcpsock_posix.go:254 +0x4d
net/http.tcpKeepAliveListener.Accept(0xc8201ec0a8, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/net/http/server.go:2135 +0x41
net/http.(*Server).Serve(0xc8201e65a0, 0x7f3ca46bc4f8, 0xc8201ec0a8, 0x0, 0x0)
	/root/go/src/net/http/server.go:1887 +0xb3
net/http.(*Server).ListenAndServe(0xc8201e65a0, 0x0, 0x0)
	/root/go/src/net/http/server.go:1877 +0x136
net/http.ListenAndServe(0xc82000aff0, 0xa, 0x7f3ca46bb460, 0xc8201d8540, 0x0, 0x0)
	/root/go/src/net/http/server.go:1967 +0x8f
main.main.func2(0xc8201e25e0)
	/root/projects/src/github.com/gophish/gophish/gophish.go:72 +0x67d
created by main.main
	/root/projects/src/github.com/gophish/gophish/gophish.go:74 +0x16d

goroutine 86 [IO wait]:
net.runtime_pollWait(0x7f3ca46bc2b8, 0x72, 0xc82000a1c0)
	/root/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820534990, 0x72, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820534990, 0x0, 0x0)
	/root/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820534930, 0xc820015000, 0x1000, 0x1000, 0x0, 0x7f3ca4efc050, 0xc82000a1c0)
	/root/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820028020, 0xc820015000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/root/go/src/net/net.go:172 +0xe4
net/http.(*liveSwitchReader).Read(0xc820336048, 0xc820015000, 0x1000, 0x1000, 0xc82011c000, 0x0, 0x0)
	/root/go/src/net/http/server.go:219 +0xa4
io.(*LimitedReader).Read(0xc8203c00a0, 0xc820015000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/root/go/src/io/io.go:427 +0xbd
bufio.(*Reader).fill(0xc82033a000)
	/root/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82033a000, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/bufio/bufio.go:132 +0xcc
net/http.(*conn).readRequest(0xc820336000, 0x0, 0x0, 0x0)
	/root/go/src/net/http/server.go:629 +0x2bb
net/http.(*conn).serve(0xc820336000)
	/root/go/src/net/http/server.go:1319 +0x727
created by net/http.(*Server).Serve
	/root/go/src/net/http/server.go:1910 +0x3f6
```

**Comment by jordan-wright** [comment]
Interesting.. Looks like something might be up with the SMTP settings - I'll take a look and will get back to you.

**Comment by Oweoqi** [comment]
No worries, I input the following into the "Host:" field: smtp.gmail.com:587

**Comment by jordan-wright** [comment]
Here's what's really slick about go. We can browse the source to track down the likely cause (**this is isn't to in any way imply that you should have done this** - just something I like to show to brag on the language :smile:)

The only place in the mail package I see the error "missing phrase" is here https://golang.org/src/net/mail/message.go#L369

This is called by the ```parseAddress``` function here: https://golang.org/src/net/mail/message.go#L267

This function is called by the "exportable" (user usable) function ```ParseAddress``` here: https://golang.org/src/net/mail/message.go#L139

We can see that the root cause is from ```worker:57``` (https://github.com/gophish/gophish/blob/a8f722b80f2791aedbec4acf447781e1e65d0596/worker/worker.go#L55), which is indeed calling ```ParseAddress```.

In a nutshell, it looks like you might be setting a misformatted "From" address in the SMTP settings. Can you confirm that you're using the ```Barry Gibbs <bg@example.com>``` format? Or just a vanilla email (I think that works, too).

**Comment by Oweoqi** [comment]
Yup that was it I just put the word test and nothing else

**Comment by jordan-wright** [comment]
Ok, awesome! Glad we could knock that out. Let us know if you run into any other issues!
